File: ObjC/3_UIView전환/SampleApp/ViewController.m
#import "ViewController.h"

@implementation ViewController

// addSubView
//     UIView
//        - 새로운 View를 추가합니다.
// removeFromSuperView
//        - 부모 뷰로부터 제거된다.

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    _childView.frame = self.view.frame;
}

- (IBAction)open:(id)sender {
  _childView.frame = CGRectMake(0, 0, 0, 0);
  _childView.transform = CGAffineTransformMakeRotation(3.14);
  // 180도 회전

  [self.view addSubview:_childView];
  
  [UIView beginAnimations:nil context:nil];
  [UIView setAnimationDuration:1];
  
  _childView.frame = self.view.frame;
  _childView.transform = CGAffineTransformMakeRotation(0);
  
  [UIView commitAnimations];
}

// Hero

- (IBAction)close:(id)sender {
  
  [UIView beginAnimations:nil context:nil];
  [UIView setAnimationDuration:1];
  
  [UIView setAnimationTransition:UIViewAnimationTransitionFlipFromLeft
                         forView:self.view
                           cache:YES];
  
  [_childView removeFromSuperview];
  
  [UIView commitAnimations];

}

@end

File: Swift/6_TableView/SampleApp/ViewController.swift
import UIKit

// open: 다른 모듈에서 클래스에 대한 확장을 허용하는 문법

class ViewController: UIViewController {
  override func viewDidLoad() {
    super.viewDidLoad()
    // Do any additional setup after loading the view.
  }
  
  // XIB
  @IBAction func tableView1(_ sender: Any) {
    let controller = TableController1()
    controller.modalPresentationStyle = .fullScreen
    present(controller, animated: true)
  }
  
  // Storyboard
  @IBAction func tableView2(_ sender: Any) {
  }
  
}

